integration-tests/tests/cases/mpc.rs

```diff
@@ -184,6 +185,195 @@ async fn test_basic_sign() {
     );
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_sign_request_retries_after_failure() {
+    const NUM_NODES: usize = 3;
+
+    let mut droppers = Vec::with_capacity(NUM_NODES);
+
+    let mut builder = MpcFixtureBuilder::default()
+        .only_generate_signatures()
+        .with_signature_timeout_ms(1_000);
+
+    for idx in 0..NUM_NODES {
+        let participant = Participant::from(idx as u32);
+        let (dropper, filter) = SignatureDropper::new(participant);
+        builder = builder.with_outgoing_message_filter(idx, filter);
+        droppers.push(dropper);
+    }
+
+    let network = builder.build().await;
+
+    for dropper in &droppers {
+        dropper.enable();
+    }
+
+    tokio::time::timeout(
+        Duration::from_millis(300),
+        network.wait_for_presignatures(2),
+    )
+    .await
+    .expect("should start with enough presignatures");
+
+    let request = sign_request(1);
+    for node in &network.nodes {
+        node.sign_tx.send(request.clone()).await.unwrap();
+    }
+
+    let start = std::time::Instant::now();
+
+    // Wait until at least one signature attempt is dropped to prove a retry occurred.
+    tokio::time::timeout(Duration::from_secs(5), async {
+        loop {
+            let dropped = droppers
+                .iter()
+                .map(SignatureDropper::dropped)
+                .sum::<usize>();
+
+            if dropped > 0 {
+                break;
+            }
+
+            tokio::time::sleep(Duration::from_millis(20)).await;
+        }
+    })
+    .await
+    .expect("expected to drop at least one signature message while filters enabled");
+
+    if start.elapsed() < Duration::from_secs(2) {
+        tokio::time::sleep(Duration::from_secs(2) - start.elapsed()).await;
+    }
+
+    for dropper in &droppers {
+        dropper.disable();
+    }
+
+    tokio::time::sleep(Duration::from_millis(100)).await;
+
+    let actions = tokio::time::timeout(Duration::from_secs(20), network.wait_for_actions(1))
+        .await
+        .expect("should publish RPC action eventually");
+
+    let dropped_messages: usize = droppers.iter().map(SignatureDropper::dropped).sum();
+    assert!(
+        actions
+            .iter()
+            .any(|action| action.contains("RpcAction::Publish")),
+        "unexpected rpc actions: {actions:?}"
+    );
+    assert!(
+        dropped_messages > 0,
+        "expected test filter to drop at least one signature message"
+    );
+    assert!(
+        start.elapsed() >= Duration::from_secs(2),
```

Comment:
What is the logic behind this check? From what I see, we are making sure that at least 2 seconds passed here:
```rust
if start.elapsed() < Duration::from_secs(2) {
        tokio::time::sleep(Duration::from_secs(2) - start.elapsed()).await;
}
```
so it must be true.

Reply:
removing this